scripts/integration-test

```diff
@@ -1,83 +1,55 @@
-while ! curl -sf http://localhost:80/healthz >/dev/null 2>&1; do
-    echo "Waiting for Rancher's /healthz endpoint to become available"
-    sleep 2
-done
+# Wait for rancher to start up
+try --delay 2 --max 30 --waitmsg "Waiting for rancher to start" --failmsg "No rancher here" kubectl rollout status --watch=true --timeout=10s -n cattle-system deploy/rancher
+echo "Rancher deployed"
 
-# Tail the rancher logs if rancher fails to deploy the webhook after 5 minutes.
-bash -c "sleep 300 && echo 'Rancher has not deployed webhook after 5m tailing logs' && tail -f ./rancher.log" &
-# Get PID of the tail command so we can kill it if needed
-TAIL_PID=$!
+# Wait for rancher to start up
+try --delay 2 --max 30 --waitmsg "Waiting for rancher/webhook to be deployed" --failmsg "No rancher here" kubectl rollout status --watch=true --timeout=10s -n cattle-system deploy/rancher-webhook
+echo "Rancher deployed"
+
+webhook_deployed() {
+    status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}") && [[ "$status" == "deployed" ]]
+}
 
 # Wait for Rancher to deploy rancher-webhook.
-while ! kubectl rollout status -w -n cattle-system deploy/rancher-webhook >/dev/null 2>&1; do
-    echo "Waiting for rancher to deploy rancher-webhook..."
-    sleep 2
-done
+try --delay 2 --max 30 --waitmsg "Waiting for webhook to be deployed (2)" webhook_deployed
 echo "Webhook deployed"
 
-# After rancher deploys webhook kill the bash command running tail.
-kill ${TAIL_PID}
-
 # Wait for helm operation to complete and save rancher-webhook release info before we kill rancher and the cluster.
-while 
-	status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}")
-	[[ "$status" != "deployed" ]]
-do 
-	echo "Waiting for helm operation to finish, current status $status"
-	sleep 2
-done
-
-# Kill Rancher since we only need the CRDs and the initial webhook values.
-# We do not want Rancher to reconcile an older version of the webhook during test.
-kill ${RANCHER_PID}
+app_catalog_deployed() {
+  status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}") && [[ "$status" == "deployed" ]]
+}
 
-echo "Rancher has been stopped starting K3s."
-# Start Cluster without Rancher.
-k3s server --cluster-init --disable=traefik,servicelb,metrics-server,local-storage --node-name=local-node --log=./k3s.log >/dev/null 2>&1 &
-KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+try --delay 2 --max 30 --waitmsg "Waiting for app-catalog to be helm-deployed (2)" app_catalog_deployed
+
+# Shut down the core rancher part, but leave the rest of the rancher environment running
 
-# Wait for cluster to start.
-while ! kubectl version >/dev/null 2>&1; do
-    echo "Waiting for cluster to start"
-    sleep 5
-done
+kubectl scale deploy rancher -n cattle-system --replicas=0
+kubectl wait pods -l app=rancher-webhook --for=delete --namespace cattle-system
```

Comment:
In the unlikely event that the rancher pod dies before the start of the `kubectl wait` call, would the script fail, wait forever or just ignore and continue?

Reply:
I added timeouts. And because `-e` is on, if a timeout is hit the script should end.